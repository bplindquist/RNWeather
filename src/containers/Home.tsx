import React from "react";
import { ScrollView, StyleSheet, Text } from "react-native";
import { useSelector } from "react-redux";
import { RootState } from "../store";

export const Home = () => {
  const { location } = useSelector((state: RootState) => ({
    location: state.locationReducer.city,
  }));

  return (
    <ScrollView
      contentContainerStyle={styles.container}
      keyboardShouldPersistTaps="handled"
    >
      <Text>{location}</Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginVertical: 24,
  },
});
