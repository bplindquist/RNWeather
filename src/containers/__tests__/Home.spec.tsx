import React from "react";
import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { Home } from "../Home";
import { createStore } from "redux";

describe("Home", () => {
  it("renders successfully", () => {
    // arrange
    const store = createStore(() => ({
      locationReducer: {
        city: "Mock",
      },
    }));

    // act
    const root = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    // assert
    expect(root.toJSON()).toMatchSnapshot();
  });
});
