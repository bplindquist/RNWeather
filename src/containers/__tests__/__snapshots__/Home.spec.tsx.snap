// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders successfully 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
      "marginVertical": 24,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <Text>
      Mock
    </Text>
  </View>
</RCTScrollView>
`;
